File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(!head) return nullptr;
        if(!head->next && n == 1) return nullptr;
        ListNode* dummy = new ListNode();
        dummy->next = head;
        ListNode* temp = head;
        for(int i=0; i<n; i++){
            temp = temp->next;
        }
        while(temp){
            temp=temp->next;
            dummy = dummy->next;
        }
        if(dummy->next == head){
            return head->next;
        }
        dummy->next = dummy->next->next;
        return head;
    }
};

File: Data Structures & Algorithms/add-two-numbers/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ans = ListNode(0)
        cur = ans
        carry = 0 
        while l1 or l2:
            sum = carry 
            if(l1):
                sum += l1.val 
                l1 = l1.next 
            if(l2):
                sum += l2.val 
                l2 = l2.next 
            cur.next = ListNode(sum%10)
            carry = sum // 10 
            cur = cur.next 
        if(carry): cur.next = ListNode(carry) 
        return ans.next